discovery-aws-api/src/main/scala/akka/discovery/awsapi/ec2/Ec2TagBasedSimpleServiceDiscovery.scala

```diff
@@ -68,6 +70,33 @@ class Ec2TagBasedSimpleServiceDiscovery(system: ActorSystem) extends SimpleServi
 
   private val runningInstancesFilter = new Filter("instance-state-name", List("running").asJava)
 
+  lazy val defaultClientConfiguration = {
+    val clientConfiguration = new ClientConfiguration()
+    // we have our own retry/back-off mechanism (in Cluster Bootstrap), so we don't need EC2Client's in addition
+    clientConfiguration.setRetryPolicy(PredefinedRetryPolicies.NO_RETRY_POLICY)
+    clientConfiguration
+  }
+
+  private val ec2Client: AmazonEC2 = {
+    val clientConfiguration = clientConfigFqcn match {
+      case Some(fqcn) ⇒
+        system.dynamicAccess.createInstanceFor[ClientConfiguration](fqcn, Nil) match {
+          case Success(clientConfig: ClientConfiguration) ⇒
+            if (clientConfig.getRetryPolicy != PredefinedRetryPolicies.NO_RETRY_POLICY) {
+              log.warning(
```

Comment:
Wonder if it would be better to fail-fast and throw an exception here instead.

Reply:
@dhpiggott I think that there are users that use this just for service discovery (and not for cluster bootstrap) hence they may want to have a configuration where a predefined retry policy is actually used. Hence a warning is a good solution for now, IMO. WDYT ?

Reply:
100% agree ; I'm changing the warning msg. a bit to make it more clear .